geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -215,49 +213,39 @@ object AccumuloDataStoreFactory {
-  val tableNameParam         = new Param("tableName", classOf[String], "Accumulo catalog table name", true)
-  val queryTimeoutParam      = GeoMesaDataStoreFactory.QueryTimeoutParam
-  val queryThreadsParam      = GeoMesaDataStoreFactory.QueryThreadsParam
-  val recordThreadsParam     = new Param("recordThreads", classOf[Integer], "The number of threads to use for record retrieval", false, 10)
-  val writeThreadsParam      = new Param("writeThreads", classOf[Integer], "The number of threads to use for writing records", false, 10)
-  val looseBBoxParam         = GeoMesaDataStoreFactory.LooseBBoxParam
-  val generateStatsParam     = GeoMesaDataStoreFactory.GenerateStatsParam
-  val collectQueryStatsParam = new Param("collectQueryStats", classOf[java.lang.Boolean], "Collect statistics on queries being run", false, true)
-  val auditQueriesParam      = GeoMesaDataStoreFactory.AuditQueriesParam
-  val cachingParam           = GeoMesaDataStoreFactory.CachingParam
-  val mockParam              = new Param("useMock", classOf[String], "Use a mock connection (for testing)", false)
-  val forceEmptyAuthsParam   = org.locationtech.geomesa.security.ForceEmptyAuthsParam
-  val NamespaceParam         = new Param("namespace", classOf[String], "Namespace", false)
+  val ConnectorParam       = new GeoMesaParam[Connector]("accumulo.connector", "Accumulo connector", deprecated = Seq("connector"))
+  val InstanceIdParam      = new GeoMesaParam[String]("accumulo.instance.id", "Accumulo Instance ID", required = true, deprecated = Seq("instanceId", "accumulo.instanceId"))
+  val ZookeepersParam      = new GeoMesaParam[String]("accumulo.zookeepers", "Zookeepers", required = true, deprecated = Seq("zookeepers"))
+  val UserParam            = new GeoMesaParam[String]("accumulo.user", "Accumulo user", required = true, deprecated = Seq("user"))
+  val PasswordParam        = new GeoMesaParam[String]("accumulo.password", "Accumulo password", metadata = Map(Parameter.IS_PASSWORD -> java.lang.Boolean.TRUE), deprecated = Seq("password"))
+  val KeytabPathParam      = new GeoMesaParam[String]("accumulo.keytab.path", "Path to keytab file", deprecated = Seq("keytabPath", "accumulo.keytabPath"))
+  val MockParam            = new GeoMesaParam[java.lang.Boolean]("accumulo.mock", "Use a mock connection (for testing)", default = false, deprecated = Seq("useMock", "accumulo.useMock"))
+  val AuthsParam           = org.locationtech.geomesa.security.AuthsParam
+  val ForceEmptyAuthsParam = org.locationtech.geomesa.security.ForceEmptyAuthsParam
+  val VisibilitiesParam    = org.locationtech.geomesa.security.VisibilitiesParam
+  val CatalogParam         = new GeoMesaParam[String]("accumulo.catalog", "Accumulo catalog table name", required = true, deprecated = Seq("tableName", "accumulo.tableName"))
+  val QueryTimeoutParam    = GeoMesaDataStoreFactory.QueryTimeoutParam
```

Comment:
Would it make sense to 'bundle' some of the params which logically go together and mix them into the [Impl]DataStoreParams objects?

Reply:
yeah, that's not a bad idea, although we'd probably only be able to bundle 2-3 of them into each trait. having them explicitly defined is kind of nice too. i dunno, what do you think?

Reply:
Well, the advantage I see to having traits is that a new ds would reuse a few handy traits (to cover namespace, security, query auditing).  

That said, we don't create too many new stores, so might not be a big deal...

Reply:
added traits for namespace, security, and general params